Fix #2750

judge2005 opened the pull request:
This is a fix for https://github.com/espressif/arduino-esp32/issues/2750

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.cpp:
```diff
@@ -1,32 +1,26 @@
+{
+}
+
+bool SPIFFSImpl::exists(const char* path) {
```
please use a new line before the curly braces. I see this a few more times below :)

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.cpp:
```diff
@@ -1,32 +1,26 @@
-// Copyright 2015-2016 Espressif Systems (Shanghai) PTE LTD
```
Do not remove the headers!

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.h:
```diff
@@ -1,39 +1,33 @@
-// Copyright 2015-2016 Espressif Systems (Shanghai) PTE LTD
```
Do not remove the headers!

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.h:
```diff
@@ -1,39 +1,33 @@
+class SPIFFSImpl : public VFSImpl
+{
+public:
+	SPIFFSImpl();
```
too much space?

me-no-dev commented:
Left a few styling notes. Not sure why you deleted the company headers?

judge2005 commented:
> Left a few styling notes. Not sure why you deleted the company headers?
I implemented these changes in separate files and then overwrote the originals. Anyway - I've added them back...

judge2005 commented:
I merged all of the recent changes into my fork before committing fixes for this. Not sure what effect that will have on attempts by you to merge this specific change to SPIFFS.

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.h:
```diff
@@ -14,26 +14,37 @@
+#pragma once
+
 #include "FS.h"
+#include "FSImpl.h"
```
these two includes should not be in the header, but in the CPP file instead. Nothing in the header depends on them :)

judge2005 replied:
The header **does** depend on them, but they are also included via vfs_api.h, so at most they are redundant. For the same reason they don't need to be included in the .cpp file (because they are included via vfs_api.h which is included via SPIFFS.h).

Whatever, I have removed them from SPIFFS.h, but I haven't added them to SPIFFS.cpp.

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.h:
```diff
@@ -14,26 +14,37 @@
 #ifndef _SPIFFS_H_
 #define _SPIFFS_H_
 
+#pragma once
```
this is redundant with the above. keep one or the other :)

me-no-dev commented on libraries/SPIFFS/src/SPIFFS.h:
```diff
@@ -14,26 +14,33 @@
 #define _SPIFFS_H_
 
-#include "FS.h"
+#include "vfs_api.h"
```
FS should be included. vfs_api is a private header.

judge2005 replied:
18 hours ago, you asked me to remove FS.h and FSImpl.h, now you are asking me to add FS.h back in.

If I remove vfs_api.h I will also have to add  FSImpl.h back because SPIFFSImpl is declared in SPIFFS.h.

me-no-dev replied:
No, I asked about `#include "FSImpl.h"` and `#include "vfs_api.h"`.
FS.h is the only one that should be included :) I guess a bit of misunderstanding.

judge2005 replied:
This change requires an additional class, SPIFFFSImpl that inherits VFSImpl, which is declared in vfs_api.h, so something is going to need to include vfs_api.h. Currently this class is declared in SPIFFS.h, which is why SPIFFS.h includes vfs_api.h. How would you have me resolve this?

me-no-dev replied:
Does this class need to be public? What happens if the declaration goes into the CPP file?

judge2005 replied:
It should be fine if I move it to the CPP file.

me-no-dev commented:
merged :)
